src/DynamoCoreWpf/Extensions/IViewExtension.cs

```diff
@@ -51,4 +51,77 @@ public interface IViewExtension: IDisposable
         /// </summary>
         void Shutdown();
     }
+
+    /// <summary>
+    /// Base class for View Extension.
+    /// </summary>
+    public abstract class ViewExtensionBaseClass : IViewExtension
+    {
+        /// <summary>
+        /// A unique id for this extension instance.  
+        /// 
+        /// There may be multiple instances of the same type, but the application 
+        /// will *not* allow two instances to coexist with the same id.
+        /// </summary>
+        public string UniqueId { get; set; }
+
+        /// <summary>
+        /// A name for the extension instance.  This is used for more user-readable logging.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Dispose method for the view extension.
+        /// </summary>
+        public void Dispose()
+        {
+           
+        }
+
+        /// <summary>
+        /// Action to be invoked when DynamoView begins to start up.  This is guaranteed to happen
+        /// after the DynamoModel has been created.
+        /// 
+        /// This method is *not* guaranteed to be invoked unless the extension is present
+        /// at startup.
+        /// 
+        /// Exceptions thrown from this method will be caught by Dynamo and 
+        /// displayed.
+        /// </summary>
+        public void Loaded(ViewLoadedParams viewLoadedParams)
+        {
+         
+        }
+
+        /// <summary>
+        /// Action to be invoked when shutdown has begun.  This gives the UI a last minute chance to interact
+        /// with the user.
+        /// 
+        /// This action is guaranteed to be invoked before the associated model layer method on IExtension.
+        /// </summary>
+        public void Shutdown()
+        {
+            
+        }
+
+        /// <summary>
+        /// Action to be invoked when DynamoView begins to start up.  This is guaranteed to happen
+        /// after the DynamoModel has been created.
+        /// 
+        /// This method is *not* guaranteed to be invoked unless the extension is present
+        /// at startup.
+        /// 
+        /// Exceptions thrown from this method will be caught by Dynamo and 
+        /// displayed.
+        /// </summary>
+        public void Startup(ViewStartupParams viewStartupParams)
+        {
+           
+        }
+
+        /// <summary>
+        /// Action to be invoked when the view extenion is closed. 
+        /// </summary>
+        public abstract void OnViewExtensionClosed(String extensionTabName);
```

Comment:
I mean it should also have empty default implementations for abstract members.

Reply:
May be I am missing something, can an abstract method have an implementation body?

Reply:
you're right, it's not possible - I'm not sure we want to use an abstract method then - maybe just virtual with a default implementation.

Reply:
I think it would be better if we followed what @mjkkirschner proposed and have something like `public virtual void Closed()` with an empty implementation. Maybe we also want to include a parameters object for future extensibility?

Reply:
Changed this.